src/main/java/com/zendesk/maxwell/filtering/Filter.java

```diff
@@ -0,0 +1,159 @@
+			String column = entry.getKey();
+
+			if (data.containsKey(column)) {
+				String expectedColumnValue = entry.getValue();
+				Object value = data.get(column);
+
+				if ("NULL".equals(expectedColumnValue)) {
+					// null or "null" (string) or "NULL" (string) is expected
+					if (value != null && !"null".equals(value) && !"NULL".equals(value)) {
+						return false;
+					}
+				} else {
+					if (value == null || !expectedColumnValue.equals(value.toString())) {
+						return false;
+					}
+				}
+			}
+		}
+
+		return true;
+	}
+
+	public boolean includes(String database, String table) {
+		FilterResult match = new FilterResult();
+
+		for ( FilterPattern p : patterns )
+			p.match(database, table, match);
+
+		return match.include;
+	}
+
+	public boolean isTableBlacklisted(String database, String table) {
+		FilterResult match = new FilterResult();
+
+		for ( FilterPattern p : patterns ) {
+			if ( p.getType() == FilterPatternType.BLACKLIST )
+				p.match(database, table, match);
+		}
+
+		return !match.include;
+	}
+
+	public boolean isDatabaseBlacklisted(String database) {
+		for ( FilterPattern p : patterns ) {
+			if (p.getType() == FilterPatternType.BLACKLIST &&
+				p.getDatabasePattern().matcher(database).find() &&
+				p.getTablePattern().toString().equals(""))
+				return true;
+		}
+
+		return false;
+	}
+
+	public static boolean isSystemBlacklisted(String databaseName, String tableName) {
+		return "mysql".equals(databaseName) &&
+			("ha_health_check".equals(tableName) || StringUtils.startsWith(tableName, "rds_heartbeat"));
+	}
+
+	public static boolean includes(Filter filter, String database, String table) {
+		if (filter == null) {
+			return true;
+		} else {
+			return filter.includes(database, table);
+		}
+	}
+
+	public static boolean matchesValues(Filter filter, Map<String, Object> data) {
+		if (filter == null) {
+			return true;
+		} else {
+			return filter.matchesValues(data);
+		}
+	}
+
+	public static Filter fromOldFormat(
+		String includeDatabases,
+		String excludeDatabases,
+		String includeTables,
+		String excludeTables,
+		String blacklistDatabases,
+		String blacklistTables,
+		String includeValues
+	) throws InvalidFilterException {
+		ArrayList<String> filterRules = new ArrayList<>();
+
+		if ( blacklistDatabases != null ) {
+			for ( String s : blacklistDatabases.split(",") )
+				filterRules.add("blacklist: " + s + ".*");
+		}
+
+		if ( blacklistTables != null ) {
+			for (String s : blacklistTables.split(","))
+				filterRules.add("blacklist: *." + s);
+		}
+
+		/* any include in old-filters is actually exclude *.* */
+		if ( includeDatabases != null || includeTables != null ) {
+			filterRules.add("exclude: *.*");
+		}
+
+		if ( includeDatabases != null ) {
+			for ( String s : includeDatabases.split(",") )
+				filterRules.add("include: " + s + ".*");
+
+		}
+
+		if ( excludeDatabases != null ) {
+			for (String s : includeDatabases.split(","))
+				filterRules.add("exclude: " + s + ".*");
```

Comment:
If it already has `exclude: *.*`, do we still want to add these duplicate rules?

Reply:
yes, because of how the old filtering worked:  you could do like:

```
include_dbs=foodb
exclude_tables=bartbl
```

which semantically meant: `exclude=*.*; include: foodb.*; exclude: *.bartbl`, ie: "only output everything in `foo` except for `bartable`"